_data-prepper/pipelines/configuration/processors/key-value.md

```diff
@@ -23,6 +23,16 @@ You can use the `key_value` processor to parse the specified field into key-valu
+| transform_key | When to change keys to lowercase, uppercase, or captalize. | If `transform_key` is `lowercase`, `{"Key1=value1"}` will parse into `{"key1": "value1"}`. <br /> If `transform_key` is `uppercase`, `{"key1=value1"}` will parse into `{"KEY1": "value1"}`. <br /> If `transform_key` is `capitalize`, `{"key1=value1"}` will parse into `{"Key1": "value1"}`. |
+| whitespace | Specify whether to be lenient or strict with the acceptance of unnecessary whitespace surrounding the configured value-split sequence. Default is `lenient`. | If `whitespace` is `"lenient"`, `{"key1  =  value1"}` will parse into `{"key1  ": "  value1"}`. If `whitespace` is `"strict"`, `{"key1  =  value1"}` will parse into `{"key1": "value1"}`. |
+| skip_duplicate_values | A Boolean option for removing duplicate key-value pairs. When set to true, only one unique key-value pair will be preserved. Default is `false`. | If `skip_duplicate_values` is `false`.,`{"key1=value1&key1=value1"}` will parse into `{"key1": ["value1", "value1"]}`. If  `skip_duplicate_values` is `true`, `{"key1=value1&key1=value1"}` will parse into `{"key1": "value1"}`. |
+| remove_brackets | Specify whether to treat square brackets, angle brackets, and parentheses as value "wrappers" that should be removed from the value. Default is `false`. | If `remove_brackets` is `true`.,`{"key1=(value1)"}` will parse into `{"key1": value1}`. If `remove_brackets` is `false`, `{"key1=(value1)"}` will parse into `{"key1": "(value1)"}`. |
+| recursive | Specify whether to drill down into values and recursively get more key-value pairs from it. The extra key-value pairs will be stored as sub-keys of the root key. Default is `false`. The levels of recursive parsing must be defined by different brackets for each level: `[]`, `()`, and `<>` in this order. Any other configurations specified will only be applied on the outmost keys. <br />While `recursive` is `true`: <br /> `remove_brackets` cannot also be `true`;<br />`skip_duplicate_values` will always be `true`; <br />`whitespace` will always be `"strict"`. | If `recursive` is true, `{"item1=[item1-subitem1=item1-subitem1-value&item1-subitem2=(item1-subitem2-subitem2A=item1-subitem2-subitem2A-value&item1-subitem2-subitem2B=item1-subitem2-subitem2B-value)]&item2=item2-value"}` will parse into `{"item1": {"item1-subitem1": "item1-subitem1-value", "item1-subitem2" {"item1-subitem2-subitem2A": "item1-subitem2-subitem2A-value", "item1-subitem2-subitem2B": "item1-subitem2-subitem2B-value"}}}`. |
```

Comment:
Confirm that my changes retain the intended meaning. Also, do we have to use "outmost", or should it be "outermost"?

Reply:
Outermost is fine.